### PR TITLE
MMO-20

### PR DESCRIPTION
**Pull request by NEON725**
Adds a basic road map document for our MVP phase.
We can use the chat function on this pull request to discuss whether you think this is an appropriate list of requirements for such a phase.

### CODE REVIEW

**Comment by NEON725** [comment]
https://neon725.atlassian.net/jira/software/projects/MMO/issues/MMO-20

File: docs/Roadmap.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Road Map
+A place to outline specific project requirements for release phase.
+Each item should eventually be annotated with a Jira ticket. When all such tickets are marked as done, the next phase can begin.
+This is not an exhaustive list of all things that will be accomplished in a phase; It is a list of minimum requirements before that phase can be released.
+Bug fixes are not described here.
+
+# MVP
+Should feature a very basic environment, and a simple avatar can jump around in that environment. [MMO-15](https://neon725.atlassian.net/jira/software/projects/MMO/issues/MMO-15)
```
Comment (VincePayne42) [inline docs/Roadmap.md:8]: At least two zones, a safe hub area and an area with basic enemies

File: docs/Roadmap.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Code should implement the entity-component-system (ECS) pattern and update the project to utilize this framework for entity behavior. [MMO-21](https://neon725.atlassian.net/jira/software/projects/MMO/issues/MMO-21)
+Project should implement multiplayer by means of a client-server model. [MMO-22](https://neon725.atlassian.net/jira/software/projects/MMO/issues/MMO-22)
```
Comment (VincePayne42) [inline docs/Roadmap.md:10]: Maybe once this is complete we perform a stress test and try and get as many client instances as possible to connect.

File: docs/Roadmap.md (hunk 1 above)
Comment (VincePayne42) [reply docs/Roadmap.md:8]: Those two things can be sharded seperately per the sharding requirements.

**Comment by VincePayne42** [comment]
We could also test having friendly NPCs as well as enemy ones, but idk how deep we want to try and enable interaction at this stage, if at all.

File: docs/Roadmap.md (hunk 2 above)
Comment (NEON725) [reply docs/Roadmap.md:10]: Agreed, but this is already covered under the tooling requirement.

**Comment by NEON725** [comment]
I do not agree that this is fitting for the first release. This starts to resemble a game mechanic when we are still focusing on framework and foundation.
I think the next phase will focus more on the core gameplay loop.

File: docs/Roadmap.md (hunk 1 above)
Comment (NEON725) [reply docs/Roadmap.md:8]: This is addressed. See new requirements added here: https://github.com/NEON725/Somnium/pull/1/commits/7efd2e2434220d7fa36a443ebe8c4c4343d057df

**Pull request merged**